tests/test.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+/* global describe, it, before, afterEach, after */
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const mockery = require('mockery');
+
+describe('Detecting a Youtube URL and transform it into a generator URL', function(){
+
+	let requestMock;
+	const youtubeURLs = {
+		video : "https://www.youtube.com/watch?v=IXxZRZxafEQ",
+		playlist : "https://www.youtube.com/watch?v=sNhhvQGsMEc&list=PLFs4vir_WsTzcfD7ZE8uO3yX-GCKUk9xZ"
+	};
+
+	let transform;
+	const host = "ftlabs-screens.herokuapp.com";
+
+	before(function () {
+		requestMock = sinon.stub();
+		mockery.registerMock('../request', requestMock);
+		mockery.registerAllowable('../client/common/js/urls.js');
+		mockery.enable({
+			useCleanCache: true
+		});
+		transform = require('../client/common/js/urls.js');
+	});
+
+	it("Should detect a Youtube video URL and create a URL to the Youtube generator specifying that it's a video", function(done){
+
+		//^(https?:\/\/[^\/]*(ftlabs-herokuapp.com))?\/generators\/youtube\?mediaURI=*([a-zA-z]{11}|[a-zA-z_-]{34})
+
+		transform(youtubeURLs.video, host).then(function(url){
```

Comment:
I would probably add some more tests to cover the Youtube features.
A test to cover Youtube playlist urls such as `https://www.youtube.com/playlist?list=PLJj06SbMAVqO6ey82DabwMNvHn-amUtOc`
And a test to cover a standard Youtube URL that doesn't get transformed - `https://www.youtube.com`